jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -704,18 +704,22 @@ public void run()
         @Override
         public void succeeded()
         {
-            HttpStream stream;
-            boolean completeStream;
+            HttpStream completeStream = null;
+            Throwable failure = null;
             try (AutoLock ignored = _lock.lock())
             {
                 assert _callbackCompleted;
                 _streamSendState = StreamSendState.LAST_COMPLETE;
-                completeStream = _handling == null;
-                stream = _stream;
+                if (_handling == null)
+                {
+                    completeStream = _stream;
+                    _stream = null;
+                    failure = _callbackFailure;
```

Comment:
We are succeeding, why looking at the failure? The old code seems more correct to me?

Reply:
I can't convince myself it will always be non-null.   I've added a check to see during testing and will remove before merging if it never triggers.